### DISCUSSION TITLE
Non-conforming HTML elements got us stuck

### DISCUSSION THREAD

**Discussion by dprothero** [Pagefind/pagefind#153] [Q&A] [answered]
Getting this error:

```
The parser has encountered a text content tag (`<style>`) in the context where it is ambiguous whether this tag should be ignored or not. And, thus, is is unclear is consequent content should be parsed as raw text or HTML markup.

This error occurs due to the limited capabilities of the streaming parsing. However, almost all of the cases of this error are caused by a non-conforming markup (e.g. a `<script>` element in `<select>` element).
```

Indeed, it is from a `<style>` tag that is nested under a `<select>` tag, but there's nothing we can do about it because it's generated by a third-party component library we need to use.

I tried adding `data-pagefind-ignore="all"` on the container with the offending HTML, but that didn't seem to help.

I've attached an example HTML that throws this error. [index_html.txt](https://github.com/CloudCannon/pagefind/files/10071812/index_html.txt)

Any ideas for a workaround?

**Comment by dprothero** [comment]
I also tried adding `data-pagefind-body` to an element after the offending HTML. No luck. :(

[index2_html.txt](https://github.com/CloudCannon/pagefind/files/10071824/index2_html.txt)

**Comment by bglw** [comment, marked as answer]
~Unfortunately there is no current way around that — Pagefind parses HTML using the https://github.com/cloudflare/lol-html library, and that error there is a parsing error passed through directly from that library. Since the HTML document itself is failing to parse, it isn't possible for Pagefind to select down to a `data-pagefind-body` or anything.~

Immediately after typing that I realized that `lol-html` can be initialized with `strict: false` to continue in the face of these parsing ambiguities. Since indexing pages for search isn't a security risk, I'll disable strict mode for Pagefind 🙌

**Reply by bglw** [reply]
Tracking as #154

**Reply by bglw** [reply]
Released as [v0.10.1](https://github.com/CloudCannon/pagefind/releases/tag/v0.10.1) 🎉

**Reply by dprothero** [reply]
Awesome :tada:
